django/db/models/fields/__init__.py

```diff
@@ -123,6 +123,8 @@ class Field(RegisterLookupMixin):
     one_to_one = None
     related_model = None
 
+    descriptor_class = DeferredAttribute
```

Comment:
Allowing this to be overridden at the class level will be really valuable for third-party apps.

Reply:
Nice! In that case, should I go ahead and document this as a feature?